exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet2/DrillParquetReader.java

```diff
@@ -233,28 +288,26 @@ public void setup(OperatorContext context, OutputMutator output) throws Executio
       this.operatorContext = context;
       schema = footer.getFileMetaData().getSchema();
       MessageType projection;
+      final List<SchemaPath> columnsNotFound = new ArrayList<>(getColumns().size());
 
       if (isStarQuery()) {
         projection = schema;
       } else {
-        columnsNotFound = new ArrayList<>();
         projection = getProjection(schema, getColumns(), columnsNotFound);
         if (projection == null) {
           projection = schema;
         }
-        if (columnsNotFound != null && columnsNotFound.size() > 0) {
-          nullFilledVectors = new ArrayList<>();
-          for (SchemaPath col: columnsNotFound) {
+        if (columnsNotFound.size() > 0) {
+          nullFilledVectors = new ArrayList<>(columnsNotFound.size());
+          for (SchemaPath col : columnsNotFound) {
             // col.toExpr() is used here as field name since we don't want to see these fields in the existing maps
             nullFilledVectors.add(
-              (NullableIntVector) output.addField(MaterializedField.create(col.toExpr(),
-                  org.apache.drill.common.types.Types.optional(TypeProtos.MinorType.INT)),
-                (Class<? extends ValueVector>) TypeHelper.getValueVectorClass(TypeProtos.MinorType.INT,
-                  TypeProtos.DataMode.OPTIONAL)));
-          }
-          if (columnsNotFound.size() == getColumns().size()) {
-            noColumnsFound = true;
+                (NullableIntVector) output.addField(MaterializedField.create(col.toExpr(),
+                    org.apache.drill.common.types.Types.optional(TypeProtos.MinorType.INT)),
+                    (Class<? extends ValueVector>) TypeHelper.getValueVectorClass(TypeProtos.MinorType.INT,
```

Comment:
This `TypeHelper.getValueVectorClass(...)` may be changed to `NullableIntVector.class`.